exploration/04_trigger_exploration.R

```diff
@@ -0,0 +1,851 @@
+    month
+  ) |> 
+  calc_metrics(
+    x = x,
+    y = asi
+  )
+
+# CHIRPS (from 2000)
+df_chirps_wy_2000 |> 
+  dplyr$mutate(
+    x = TRUE
+  ) |> 
+  tidyr$complete(
+    year = 2000:2024,
+    month = 1:5,
+    fill = list(
+      indicator = "CHIRPS",
+      x = FALSE
+    )
+  ) |> 
+  dplyr$left_join(
+    df_val,
+    by = "year"
+  ) |>
+  dplyr$group_by(
+    month
+  ) |> 
+  calc_metrics(
+    x = x,
+    y = asi_2000
+  )
+
+
+# SNOW
+
+df_snow_perf <- df_snow_wy |> 
+  dplyr$mutate(
+    x = TRUE
+  ) |> 
+  tidyr$complete(
+    year = 2000:2024,
+    month = c(11:12, 1:3),
+    fill = list(
+      indicator = "SWE",
+      x = FALSE
+    )
+  ) |> 
+  dplyr$left_join(
+    df_val,
+    by = "year"
+  ) |>
+  dplyr$group_by(
+    month
+  ) |> 
+  calc_metrics(
+    x = x,
+    y = asi_2000
+  )
+
+# ASI
+
+df_asi_perf <- df_asi_dek_wy |> 
+  dplyr$mutate(
+    x = TRUE
+  ) |> 
+  tidyr$complete(
+    year = 1984:2024,
+    month = 1:5,
+    dekad = 1:3,
+    fill = list(
+      indicator = "ASI",
+      x = FALSE
+    )
+  ) |> 
+  dplyr$left_join(
+    df_val,
+    by = "year"
+  ) |>
+  dplyr$group_by(
+    month,
+    dekad
+  ) |> 
+  calc_metrics(
+    x = x,
+    y = asi
+  )
+
+# Combo
+df_combo_perf <- df_combo_wy |> 
+  dplyr$ungroup() |> 
+  dplyr$mutate(
+    x = TRUE,
+    indicator = "CHIRPS\n& SWE"
+  ) |> 
+  tidyr$complete(
+    year = 2000:2024,
+    month = 1:5,
+    fill = list(
+      indicator = "CHIRPS\n& SWE",
+      x = FALSE
+    )
+  ) |> 
+  dplyr$left_join(
+    df_val,
+    by = "year"
+  ) |> 
+  dplyr$group_by(month) |> 
+  calc_metrics(
+    x = x,
+    y = asi_2000
+  )
+
+# NDVI
+
+df_ndvi_perf <- df_ndvi_wy |> 
+  dplyr$mutate(
+    x = TRUE
+  ) |> 
+  tidyr$complete(
+    year = 1984:2024,
+    month = 3:5,
+    dekad = 1:3,
+    fill = list(
+      indicator = "NDVI",
+      x = FALSE
+    )
+  ) |> 
+  dplyr$left_join(
+    df_val,
+    by = "year"
+  ) |>
+  dplyr$group_by(
+    month,
+    dekad
+  ) |> 
+  calc_metrics(
+    x = x,
+    y = asi
+  )
+
+###########################
+#### RAINFALL TIMELINE ####
+###########################
+
+df_chirps |>
+  dplyr$mutate(
+    month = factor(months(date, abbreviate = TRUE), levels = month.abb)
+  ) |>
+  gg$ggplot(
+    mapping = gg$aes(x = month, y = value)
+  ) +
+  gg$geom_boxplot() +
+  gg$labs(
+    x = "",
+    y = "Precipitation (mm)",
+    title = "Distribution of monthly rainfall, 1984 to 2024, CHIRPS"
+  )
+
+###########################
+#### JOINT PERFORMANCE ####
+###########################
+
+dplyr$bind_rows(
+  df_chirps_perf,
+  df_seas5_perf |> dplyr$rename(month = pub_month),
+  df_ndvi_perf |> dplyr$filter(dekad == 3) |> dplyr$select(-dekad), # last dekad best performance
+  df_snow_perf,
+  df_combo_perf,
+  df_asi_perf |> dplyr$filter(dekad == 3) |> dplyr$select(-dekad), # last dekad
+) |> 
+  dplyr$mutate(
+    month_label = factor(month.name[month], levels = month.name[c(11:12, 1:5)])
+  ) |> 
+  gg$ggplot(
```

Comment:
beautiful, after going through the code it makes perfect sense for me, but could probably use a more descriptive title for less informed audience to whom it will be presented. I think it's been shared around to some degree, but we can still adjust for future iterations

just 1 quick rough idea:
Title: "How well can we predict the worst drought years as measured by ASIS"
subtitle: "ASIS values taken at end of spring wheat season" (maybe subtitle unecessary).
caption: "Methodology compares how well a 1 in 5 year RP level threshold  correlates to the the worst to 1 in 5 year level drought at the end of the spring wheat season as measured by ASIS